src/mono/wasm/runtime/memory.ts

```diff
@@ -65,43 +68,58 @@ export function setB32(offset: MemOffset, value: number | boolean): void {
 
 export function setU8(offset: MemOffset, value: number): void {
     assert_int_in_range(value, 0, 0xFF);
+    updateGrowableHeapViews();
     Module.HEAPU8[<any>offset] = value;
 }
 
 export function setU16(offset: MemOffset, value: number): void {
+    assert_int_in_range(value, 0, 0xFFFF);
+    updateGrowableHeapViews();
+    Module.HEAPU16[<any>offset >>> 1] = value;
+}
+
+// does not check for growable heap
+export function setU16_local(offset: MemOffset, value: number): void {
```

Comment:
it may make sense for the _local functions to actually accept the heap as an argument, so you can't use them without having previously grabbed it